backend/tests/clubs/test_views.py

```diff
@@ -1620,6 +1622,99 @@ def test_club_invite_email_resend(self):
         )
         self.assertIn(resp.status_code, [200, 201], resp.content)
 
+    def test_send_invite_with_acceptance_email(self):
+        self.client.login(username=self.user1.username, password="test")
+
+        # Create a club application
+        now = timezone.now()
+        application = ClubApplication.objects.create(
+            name="Ordinary Name",
+            club=self.club1,
+            application_start_time=now,
+            application_end_time=now + timezone.timedelta(days=1),
+            result_release_time=now + timezone.timedelta(days=2),
+            is_wharton_council=False,
+        )
+        application.save()
+
+        applicants = [self.user1, self.user2, self.user3, self.user4, self.user5]
+        statuses = [
+            ApplicationSubmission.ACCEPTED,
+            ApplicationSubmission.ACCEPTED,
+            ApplicationSubmission.REJECTED_AFTER_INTERVIEW,
+            ApplicationSubmission.REJECTED_AFTER_WRITTEN,
+            ApplicationSubmission.PENDING,
+        ]
+
+        # Create submissions for each applicant
+        ApplicationSubmission.objects.bulk_create(
+            [
+                ApplicationSubmission(
+                    user=applicant,
+                    application=application,
+                    committee=None,
+                    status=status,
+                    reason="a reason",
+                )
+                for applicant, status in zip(applicants, statuses)
+            ]
+        )
+
+        # Send acceptance & invitation emails
+        resp = self.client.post(
+            reverse(
+                "club-applications-send-emails",
+                kwargs={"club_code": self.club1.code, "pk": application.pk},
+            ),
+            {
+                "allow_resend": True,
+                "dry_run": False,
+                "email_type": {"id": "acceptance", "name": "Acceptance"},
+            },
+            content_type="application/json",
+        )
+        self.assertIn(resp.status_code, [200, 201], resp.content)
+
+        # Check that invitations were created
+        invite1 = MembershipInvite.objects.filter(
+            club=application.club, email=self.user1.email
+        ).first()
+        invite2 = MembershipInvite.objects.filter(
+            club=application.club, email=self.user2.email
+        ).first()
+        self.assertIsNotNone(invite1)
+        self.assertIsNotNone(invite2)
+
+        expiry_date = now + timezone.timedelta(days=5)
+
+        # Compare the expiry dates excluding microseconds
+        self.assertEqual(
+            invite1.expires_at.replace(microsecond=0),
+            expiry_date.replace(microsecond=0),
+        )
+
+        self.assertEqual(
+            invite2.expires_at.replace(microsecond=0),
+            expiry_date.replace(microsecond=0),
+        )
```

Comment:
This should check for a range, instead of exact equality. (Something like "check if the invite expires within 5 minutes plus/minus of the expiry date".) Unit tests should be deterministic. We don't want this failing if the invites get sent out at 0.99 seconds, and the assert statement runs at 1.01.

Reply:
This ended up causing test failures. Flagged in #662.